README.md

```diff
@@ -41,33 +41,56 @@ instead of
-
 ### Top-Level Constructs
+The following is a partial EBNF for the top-level constructs in Ecmarkdown.
+
+```
+Document = Paragraph , [ { "\n\n" , Paragraph  } ] ;
 
-An Ecmarkdown **document**, processed using the `ecmarkdown.document(string)` export, is composed of multiple pararaphs.
+Paragraph = OrderedList | UnorderedList | NonList ;
 
-A **paragraph** can be either a list or non-list paragraph.
+OrderedList = OrderedListItem , [ { "\n" , OrderedListItem | UnorderedListItem } ] ;
 
-A **list** is composed of one or more **list items**, which are segments. Non-list paragraphs are also segments.
+UnorderedList = UnorderedListItem , [ { "\n" , OrderedListItem | UnorderedListItem } ] ;
 
-A **segment** is a list of non-formatting characters (i.e., literal text) and formatting characters.
+OrderedListItem = { white space } , { digit } , "." , space , Fragment ;
 
-An Ecmarkdown **fragment**, processed using the `ecmarkdown.fragment(string)` export, is a segment that is assumed to contain no HTML. Thus, you can pass in `` "`2 < 3`" `` as a fragment, and you'll get back as the output HTML `"<code>2 &lt; 3</code>"`.
+UnorderedListItem = { white space } , "*" , space , Fragment ;
 
-Ecmarkdown also allows HTML comments anywhere that does not otherwise disturb the parsing. They are included in the resulting output, in the equivalent location.
+NonList = Fragment ;
+
+Fragment = { StarFormat | UnderscoreFormat | TildeFormat
+         | TickFormat | PipeFormat | Text } ;
+         // could consider renaming to output element
+
+StarFormat = "*" , Text , "*" ;
+// With some complex restrictions on surrounding context to match gmd semantics
+
+// and etc. for each fragment type except TickFragment which has no context
+// restrictions per gmd semantics and UnderscoreFormat which doesn't allow whitespace
+// or format characters inside of it.
+```
+
+The key takeaways from this grammar are the following:
+
+* Paragraphs are separated by two line breaks
+* Paragraphs can either be lists or non-lists.
+* Lists can be ordered or unordered. List items must be preceded by a line break. You can have a ordered list as a sublist in an unordered list and vice versa.
+* List items and non-list paragraphs are Fragments. Fragments are a series of one or more formats or plain text.
+
+Ecmarkdown also allows HTML tags and comments anywhere that does not otherwise disturb the parsing. They are included in the resulting output, in the equivalent location.
 
 ### Lists
 
-Lists are written as a series of lines, each starting with a number, e.g. `1.`. The first list item's number determines the starting number in the output (via `<ol start="x">`); subsequent list items' numbers are ignored.
+Lists are written as a series of lines, each starting with either a number, e.g. `1.`, or a star, e.g. `*`. The first list item's number determines the starting number in the output (via `<ol start="x">`); subsequent list items' numbers are ignored.
```

Comment:
If (as indicated in issue #28) `Text` can contain linebreaks (as long as it isn't two adjacent ones), then the grammar would allow the following as an `OrderedList`:

    1. This is the first item,
       which I've split into two lines.
    2. This is the second item,
       which is
       three lines long.

However, that conflicts with the statement here that each line of a list starts with either a number or a star.

Maybe instead, each list is a series of *items*, each starting with etc.